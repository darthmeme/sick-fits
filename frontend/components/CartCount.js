import React from 'react'
import styled from 'styled-components'
import { TransitionGroup, CSSTransition } from 'react-transition-group'

const Dot = styled.div`
  background: ${props => props.theme.red};
  color: white;
  line-height: 2rem;
  min-width: 3rem;
  border-radius: 50%;
  padding: 0.5rem;
  margin-left: 1rem;
  font-weight: 400;
  font-feature-settings: "tnum";
  font-variant-numeric: tabular-nums;
`

const AnimationStyles = styled.span`
  position: relative;
  
  .count {
    display: block;
    position: relative;
    backface-visibility: hidden;
    transition: all 0.3s ease;
  }

  .count-enter {
    transform: rotateX(0.5turn)
  }

  .count-enter-active {
    transform: rotateX(0)
  }

  .count-exit {
    top: 0;
    position: absolute;
    transform: rotateX(0);
  }

  .count-exit-active {
    transform: rotateX(0.5turn);
  }
`

const CartCount = ({ count }) => (
  <AnimationStyles>
    <TransitionGroup>
      <CSSTransition
        unmountOnExit
        className="count"
        classNames="count"
        key={count}
        timeout={{ enter: 300, exit: 300 }}>
        <Dot>{count}</Dot>
      </CSSTransition>
    </TransitionGroup>
  </AnimationStyles>
)

export default CartCount
