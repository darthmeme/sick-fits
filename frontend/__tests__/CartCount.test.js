import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import CartCount from '../components/CartCount'

describe('<CartCount />', () => {
  it('renders', () => {
    shallow(<CartCount count={10} />)
  })

  it('matches the snapshot', () => {
    const wrapper = shallow(<CartCount count={5} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('can update via props', () => {
    const wrapper = shallow(<CartCount count={1} />)
    expect(toJSON(wrapper)).toMatchSnapshot()

    wrapper.setProps({ count: 2 })
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
