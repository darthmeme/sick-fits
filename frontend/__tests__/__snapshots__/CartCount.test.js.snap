// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> can update via props 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="1"
      timeout={
        Object {
          "enter": 300,
          "exit": 300,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        1
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCount /> can update via props 2`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="2"
      timeout={
        Object {
          "enter": 300,
          "exit": 300,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        2
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCount /> matches the snapshot 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="5"
      timeout={
        Object {
          "enter": 300,
          "exit": 300,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        5
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;
